SU2_CFD/include/numerics/turbulent/turb_sources.hpp

```diff
@@ -320,12 +322,61 @@ class CSourcePieceWise_TurbSST final : public CNumerics {
 
   bool incompressible;
   bool sustaining_terms;
+  bool axisymmetric;
 
   /*!
    * \brief A virtual member. Get strain magnitude based on perturbed reynolds stress matrix
    * \param[in] turb_ke: turbulent kinetic energy of the node
    */
   void SetPerturbedStrainMag(su2double turb_ke);
+  
+  /*!
+   * \brief Add contribution due to axisymmetric formulation to 2D residual
+   */
+  inline void ResidualAxisymmetric(su2double alfa_blended, su2double zeta){
+
+    if (Coord_i[1] < EPS) {
+      return;
+    }
+    
+    else{
+      
+      su2double yinv, rhov;
+      su2double sigma_k_i, sigma_omega_i;
+      su2double pk_axi, pw_axi, ck_axi, cw_axi, dk_axi, dw_axi;
+      
+      yinv = 1.0/Coord_i[1];
+      rhov = Density_i*V_i[2];
+      
+      /*--- Compute blended constants ---*/
+      sigma_k_i = F1_i*sigma_k_1 + (1.0 - F1_i)*sigma_k_2;
+      sigma_omega_i = F1_i*sigma_omega_1 + (1.0 - F1_i)*sigma_omega_2;
+  
+      /*--- Production ---*/
+      pk_axi = max(0.0,2.0/3.0*rhov*TurbVar_i[0]*(2.0/zeta*(yinv*V_i[2]-PrimVar_Grad_i[2][1]
+                                                                  -PrimVar_Grad_i[1][0]) -1.0));
+      pw_axi = alfa_blended*zeta/TurbVar_i[0]*pk_axi;
+      
+      /*--- Convection ---*/
+      ck_axi = rhov*TurbVar_i[0];
+      cw_axi = rhov*TurbVar_i[1];
+      
+      /*--- Diffusion ---*/
+      dk_axi = (Laminar_Viscosity_i+sigma_k_i*Eddy_Viscosity_i)*TurbVar_Grad_i[0][1];
+      dw_axi = (Laminar_Viscosity_i+sigma_omega_i*Eddy_Viscosity_i)*TurbVar_Grad_i[1][1];
+      
+      /*--- Add all terms to the residuals ---*/
+      Residual[0] += yinv*Volume*(pk_axi-ck_axi+dk_axi);
+      Residual[1] += yinv*Volume*(pw_axi-cw_axi+dw_axi);
+    
+      /*--- Add contribution to the jacobian for implicit time integration---*/
+      Jacobian_i[0][0] += yinv*Volume*(sigma_k_i/zeta*TurbVar_Grad_i[0][1]-V_i[2]);
+      Jacobian_i[0][1] -= yinv*Volume*sigma_k_i*TurbVar_i[0]*TurbVar_Grad_i[0][1]/(zeta*zeta);
+      Jacobian_i[1][0] += yinv*Volume*sigma_k_i/zeta*TurbVar_Grad_i[1][1];
+      Jacobian_i[1][1] -= yinv*Volume*(sigma_k_i*TurbVar_i[0]*TurbVar_Grad_i[1][1]/(zeta*zeta)+V_i[2]);
```

Comment:
All the terms added to the normal SST Jacobian are negative to avoid imbalancing the diagonal dominance of the matrix.
Do you have a feel for how large the axi symmetric contributions might be compared with the normal ones? (I guess larger near the axis).

Reply:
They may have the same magnitude as the normal ones and may be negative or positive depending on the sign of the y velocity and the gradients. Yes the further from the axis the more the problem becomes just normal 2D. I will try different versions with nothing more added to the jacobian or only convection or only diffusion (or setting anything positive to zero perhaps but that seems no good idea).

Reply:
The "clamping things with min/max" to have the sign you want is very common.
We have a config option for "accurate flux Jacobians" that we use for other schemes, you can also base the decision to clamp or not on that option to make things more flexible.